test/e2e/functional/tests/e2e.go

```diff
@@ -1085,70 +939,99 @@ var _ = Describe("metallb", func() {
 - name: addresspool1
   protocol: layer2
   addresses:
   - 1.1.1.1-1.1.1.100
-`))
+  `)
 
 			By("Deleting the Configmap")
 			err = testclient.Client.ConfigMaps(OperatorNameSpace).Delete(context.Background(), consts.MetalLBConfigMapName, metav1.DeleteOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() (string, error) {
-				var err error
-				configMap, err = testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				if err != nil {
-					return "", err
-				}
-				return configMap.Data[consts.MetalLBConfigMapName], err
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML(`address-pools:
+			assertConfigMapMatchYAML(`address-pools:
 - name: addresspool1
   protocol: layer2
   addresses:
   - 1.1.1.1-1.1.1.100
-`))
+`)
 
 			By("Checking AddressPool resource is deleted and ConfigMap is cleared")
 			err = testclient.Client.Delete(context.Background(), addressPool)
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() string {
-				configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return configmap.Data[consts.MetalLBConfigMapName]
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML("{}"))
+			assertConfigMapMatchYAML("{}")
 		})
 	})
 })
+
+func isValidationWebhookEnabled() bool {
+	deploy, err := testclient.Client.Deployments(OperatorNameSpace).Get(context.Background(), consts.MetalLBOperatorDeploymentName, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	isEnabled := false
+	for _, container := range deploy.Spec.Template.Spec.Containers {
+		if container.Name == "manager" {
+			for _, env := range container.Env {
+				if env.Name == "ENABLE_OPERATOR_WEBHOOK" {
+					if env.Value == "true" {
+						isEnabled = true
+					}
+				}
+			}
+		}
+	}
+
+	return isEnabled
+}
+
+func isValidationWebhookRunning(validationWebhookName string) bool {
+	// Can't just check the ValidatingWebhookConfiguration name as it's changing between different deployment methods.
+	// Need to check the webhook name in the webhooks definition of the ValidatingWebhookConfiguration.
+	validateCfgList := &admv1.ValidatingWebhookConfigurationList{}
+	err := testclient.Client.List(context.TODO(), validateCfgList, &goclient.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	isRunning := false
+	for _, validateCfg := range validateCfgList.Items {
+		for _, webhook := range validateCfg.Webhooks {
+			if webhook.Name == validationWebhookName {
+				isRunning = true
+			}
+		}
+	}
+
+	return isRunning
+}
+
+func assertConfigMapMatchYAML(yaml string) {
+	EventuallyWithOffset(1, func() string {
+		configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
```

Comment:
I'd move this out of the if and remove the external one, so the expect can cover the err not nil case

Reply:
good idea.